include/rocksdb/table.h

```diff
@@ -79,19 +127,45 @@ struct BlockBasedTableOptions {
   // than data blocks.
   bool cache_index_and_filter_blocks_with_high_priority = true;
 
+  // DEPRECATED: This option will be removed in a future version. For now, this
+  // option still takes effect on relevant `CachePinningOptions` members that
+  // are set to `PinningTier::kFallback`. In particular, the overrides from
+  // setting this option (and the changes users should make to migrate away from
+  // setting this option) are:
+  //
+  // - `metablock_partition_pinning == PinningTier::kFallback` ->
+  //   `metablock_partition_pinning == PinningTier::kMaybeFlushed`
+  // - `unpartitioned_metablock_pinning == PinningTier::kFallback` ->
+  //   `unpartitioned_metablock_pinning == PinningTier::kMaybeFlushed`
+  //
   // if cache_index_and_filter_blocks is true and the below is true, then
   // filter and index blocks are stored in the cache, but a reference is
   // held in the "table reader" object so the blocks are pinned and only
   // evicted from cache when the table reader is freed.
   bool pin_l0_filter_and_index_blocks_in_cache = false;
 
+  // DEPRECATED: This option will be removed in a future version. For now, this
+  // option still takes effect on relevant `CachePinningOptions` members that
+  // are set to `PinningTier::kFallback`. In particular, the override from
+  // setting this option (and the change users should make to migrate away from
+  // setting this option) is:
+  //
+  // - `metablock_top_level_index_pinning == PinningTier::kFallback` ->
+  //   `metablock_top_level_index_pinning == PinningTier::kAll`
+  //
```

Comment:
The same comment applies as above.  The default for the PinningOption should be whatever the behavior for "true" means.  If this value is "false" and the PinningOption is the default, turn off the PinningOption.